sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/ParquetCachedBatchSerializer.scala

```diff
@@ -907,8 +907,10 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
           index -> inMemReqSparkSchema.fields.indexOf(reqSparkSchemaInCacheOrder.fields(index))
         }.toMap
 
-        val reqParquetSchemaInCacheOrder =
-          sparkToParquetSchemaConverter.convert(reqSparkSchemaInCacheOrder)
+        val reqParquetSchemaInCacheOrder = new org.apache.parquet.schema.MessageType(
```

Comment:
it would be good to break up L870-956 code in smaller methods.

Have a method/class encapsulating SparkToParquetSchemaConverter and handling both cases of StructType to Parquet schema conversions with a comment

Reply:
If this is just a NIT, can we do this as a follow-on please?

Reply:
not a nit, now we use two different methods to map a Spark schema to Parquet without any explanation. If we had a dedicated method for this it would be easy to put a comment explaining this, and the code would be more readable.

Reply:
I have removed the ParquetToSparkSchemaConverter as I agree it was causing confusion. PTAL and let me know if this is more clear.